base/strings/util.jl

```diff
@@ -45,9 +45,12 @@ function endswith(a::AbstractString, b::AbstractString)
 end
 endswith(str::AbstractString, chars::Chars) = !isempty(str) && last(str) in chars
 
-# FIXME: check that end of `b` doesn't match a partial character in `a`
-startswith(a::String, b::String) = sizeof(a) ≥ sizeof(b) &&
-    ccall(:memcmp, Int32, (Ptr{UInt8}, Ptr{UInt8}, UInt), a, b, sizeof(b)) == 0
+function startswith(a::String, b::String)
+    cub = ncodeunits(b)
+    (ncodeunits(a) < cub || ccall(:memcmp, Int32, (Ptr{UInt8}, Ptr{UInt8}, UInt),
+                                   a, b, sizeof(b)) ≠ 0) && return false
```

Comment:
Missing space for alignment. Would probably be more readable as a `if`.

Reply:
rewritten to use `if` now.